docs/Troubleshooting.md

```diff
@@ -34,3 +34,39 @@ any issues
 1. Send screenshots to [Quilt support](mailto:support@quiltdata.io).
 1. It is not recommended that you adjust ElasticSearch via Edit domain, as these
 changes will be lost the next time that you update Quilt
+
+## Missing metadata when working with data packages via the API
+
+> `Package.set_dir()` on the package root (".") overrides package-level metadata.
+> If you do not provide `set_dir(".", foo, meta=baz)` with a value for `meta=`,
+> `set_dir` will set package-level metadata to `None`.
+
+A common pattern is to `Package.browse()` to get the most recent
+version of a package, and then `Package.push()` updates.
+You can preserve package-level metadata when calling `set_dir(".", ...)`
+as follows:
+
+<!--pytest.mark.skip-->
+```python
+import quilt3
+
+p = quilt3.Package.browse(
+    f"user-packages/{package_name}", 
+    f"s3://{bucket}"
+)
+
+p.set_dir(
+    ".",
+    f"s3://{bucket}/user-packages/{package_name}/",
```

Comment:
Wait the right hand side of this setdir is super duper confusing and wrong. Use “s3://somewhere/else” (not literally that but you get the idea).

Reply:
I’d recommend much simpler strings in general as well